Wangscape/tilegen/alpha/CalculatorTopTwo.cpp

```diff
@@ -0,0 +1,54 @@
+#include "CalculatorTopTwo.h"
+
+#include <algorithm>
+#include <cassert>
+
+namespace tilegen
+{
+namespace alpha
+{
+
+
+void CalculatorTopTwo::updateAlphasAux(const Weights& weights)
+{
+    initialiseIndexedWeights(weights);
+    std::sort(mIndexedWeights.begin(), mIndexedWeights.end(), std::greater<IndexedWeight>());
+    const IndexedWeight& winner = mIndexedWeights[0];
+    const IndexedWeight& runner_up = mIndexedWeights[1];
+    const IndexedWeight& base = mIndexedWeights[2];
+
+    double gap_1 = winner.first - runner_up.first;
+    double gap_2 = runner_up.first - base.first;
```

Comment:
I wonder if they're needed. `total_gap` is equal to `winner.first - base.first` and it seems that `gap_1` and `gap_2` exist only for assertions. If this is the case, you could remove them, I think.